### PR TITLE
added if to check if new variable has been added

### LINKED ISSUE #12144: Error when saving dashboard after creating first template variable
*Steps*
1. Add new dashboard, new panel, save.
2. Add first template variable, then save.
 -> It will ask if leaving the page is OK, in any case the new variable wont be saved.

*Additional info*

`SaveDashboardModalCtrl.compareTemplating` is iterating over the old templates *and accessing* them, which throws an error in the console:
```
angular.js?10a4:14700 TypeError: Cannot read property 'current' of undefined
    at SaveDashboardModalCtrl.compareTemplating
```

### PR DESCRIPTION
**Pull request by Ijin08**
fixes #12144

### CODE REVIEW

File: public/app/features/dashboard/save_modal.ts [hunk 1]
```diff
@@ -91,6 +91,9 @@ export class SaveDashboardModalCtrl {
   }
 
   compareTemplating() {
+    if (this.dashboardSrv.dash.originalTemplating.length !== this.dashboardSrv.dash.templating.list.length) {
+      return (this.variableChange = false);
```
Comment (davkal) [inline public/app/features/dashboard/save_modal.ts]: If the lists are not the same length, I would have assumed `variableChange` to be true. Unless `variableChange` has a different semantic.

**Review by davkal — CHANGES_REQUESTED** [review]
Different lengths set `variableChange` to false. If that's intended it needs a comment about its semantics.

File: public/app/features/dashboard/save_modal.ts (hunk 1 above)
Comment (Ijin08) [reply public/app/features/dashboard/save_modal.ts]: The variable change refers to changing variable while viewing and not adding and removing variables, I will add a comment that explains it in a good way.

Comment (Ijin08) [reply public/app/features/dashboard/save_modal.ts]: @davkal maybe changing name to variableValueChange would make it more clear

**Comment by davkal** [comment]
Variable needs to be renamed in test too.

**Review by davkal — APPROVED** [review]

**Pull request merged**